1.14.4/x86_64-bionic/Dockerfile

```diff
@@ -1,7 +1,55 @@
-FROM ubuntu:bionic
+FROM ubuntu:bionic AS verify
 
-# To improve : static hash make dynamic build of versions impossible.
-ARG VERSION=1.14.4
+WORKDIR /verify
+
+# Specify release variables
+ARG RLS_VERSION=1.14.4
+ARG RLS_OS=linux
+ARG RLS_LIB=gnu
+ARG RLS_ARCH=
+
+# Automatically detect architecture
+RUN set -ex && ARCHITECTURE=$(dpkg --print-architecture) \
+    && if [ "${ARCHITECTURE}" = "amd64" ]; then RLS_ARCH=x86_64 ; fi \
+    && if [ "${ARCHITECTURE}" = "arm64" ]; then RLS_ARCH=aarch64; fi \
+    && if [ "${ARCHITECTURE}" = "armhf" ]; then RLS_ARCH=arm && RLS_LIB=gnuabihf; fi \
+    && if [ "${ARCHITECTURE}" = "i386" ]; then RLS_ARCH=i686-pc; fi \
+    && if [ "${RLS_ARCH}" = "" ]; then echo "Could not determine architecture"; exit 1; fi \
+    && RLS_FILE_NAME=dogecoin-${RLS_VERSION}-${RLS_ARCH}-${RLS_OS}-${RLS_LIB}.tar.gz \
+    && echo -n ${RLS_FILE_NAME} > .filename
+
+ARG SIG_PATH=${RLS_VERSION}-${RLS_OS}
+ARG DESCRIPTOR_PATH=dogecoin/contrib/gitian-descriptors/gitian-${RLS_OS}.yml
+
+ARG RLS_LOCATION=https://github.com/dogecoin/dogecoin/releases/download/v${RLS_VERSION}
+ARG REPO_GITIAN_BUILDER=https://github.com/devrandom/gitian-builder.git
+ARG REPO_GITIAN_SIGS=https://github.com/dogecoin/gitian.sigs.git
+ARG REPO_DOGECOIN_CORE=https://github.com/dogecoin/dogecoin.git
+
+# install system requirements
+RUN apt update && apt install -y \
+    wget \
+    git \
+    ruby \
+    gpg \
+    && rm -rf /var/lib/apt/lists/*
+
+# fetch tools and setup signers
+RUN git clone --depth 1 ${REPO_GITIAN_BUILDER} gitian && \
+    git clone --depth 1 ${REPO_GITIAN_SIGS} sigs && \
+    git clone --depth 1 -b v${RLS_VERSION} ${REPO_DOGECOIN_CORE} dogecoin && \
+    find dogecoin/contrib/gitian-keys -name "*.pgp" |xargs -n 1 gpg --import
+
+# download release binary and verify against random signer
+RUN RLS_FILE_NAME=`cat .filename` && \
+    wget ${RLS_LOCATION}/${RLS_FILE_NAME} && \
+    gitian/bin/gverify --no-markup -d sigs -r ${SIG_PATH} \
+      ${DESCRIPTOR_PATH} | grep OK | \
+      shuf -n 1 | sed s/:.*// > random_signer.txt && \
+    grep ${RLS_FILE_NAME} sigs/${SIG_PATH}/$(cat random_signer.txt)/*assert | shasum -c && \
+    mv ${RLS_FILE_NAME} dogecoin.tar.gz
```

Comment:
At the cost of having a clear separation of declaration and execution, yes. Does the file bother you?

Reply:
I see. Are you maybe following a coding practice which do not apply to Dockerfile ? Separation of declaration and execution is not the rule in Dockerfile.

For exemple, [php](https://github.com/docker-library/php/blob/master/8.0/buster/apache/Dockerfile) & [golang](https://github.com/docker-library/golang/blob/master/1.17/bullseye/Dockerfile) are mixing declaration and execution. And probably all other Docker official images.

It's even a recommendation to use the cache with more efficiency, from [cacheability](https://github.com/docker-library/official-images#cacheability) recommendation:
> Ensure that lines which are less likely to change come before lines that are more likely to change (with the caveat that each line should generate an image that still runs successfully without assumptions of later lines).
>
> For example, the line that contains the software version number (`ENV MYSOFTWARE_VERSION 4.2`) should come after a line that sets up the APT repository `.list` file (`RUN echo 'deb http://example.com/mysoftware/debian some-suite main' > /etc/apt/sources.list.d/mysoftware.list`).

Doing declarations at the top of a Dockerfile would break the utility of caching mechanism.

Reply:
What about this previous comment ? I was more in favor of doing all at once, I thought at first it was a matter of taste and wouldn't complain about your arguments of separation of declaration & execution.

But this separation isn't appropriate for Dockerfiles, it's not done anywhere else (at least in official images), it's against best practices, and it's making cache unusable.

Let's drop this idea of ​​separating the declaration and the execution.

Reply:
I of course agree with the recommendation that getting more cache hits gets enhanced by doing things in the order of the least changed execution. However, I don't think we have done the analysis properly of what changes more often from a resulting image perspective, and in which scenarios we absolutely need a cache wipe. Note that the entire verification stage doesn't contain a `COPY` command. So let's do it separately. It's not just this statement.

Reply:
I've recorded this in #22